### PR TITLE
RP2 writev/readv implementation

### PR DESCRIPTION
**Pull request by ikskuh**
I've started implementing the writev/readv parts of the code and did some chores to clean up the code on the go

Not mergable yet as SPI devices are missing. Will add them and do some more cleanups before we're ready to merge

### CODE REVIEW

**Review by haydenridd — APPROVED** [review]
Looks great to me! Appreciate some of the cleanup surrounding repeated code in I2C :)

File: port/raspberrypi/rp2xxx/src/hal/i2c.zig [hunk 1]
```diff
@@ -321,46 +340,70 @@ pub const I2C = enum(u1) {
...
+    ///       This pattern allows one to create better zero-copy send routines as message prefixes and
+    ///       suffixes won't need to be concatenated/inserted to the original buffer, but can be managed
+    ///       in a separate memory.
+    ///
+    pub fn writev_blocking(i2c: I2C, addr: Address, chunks: []const []const u8, timeout: ?time.Duration) TransactionError!void {
+        if (addr.is_reserved())
+            return TransactionError.TargetAddressReserved;
+
+        const write_vec = microzig.utilities.Slice_Vector([]const u8).init(chunks);
+        if (write_vec.len() == 0)
```
Comment (haydenridd) [inline port/raspberrypi/rp2xxx/src/hal/i2c.zig:388]: Needs to be `size()`!

File: port/raspberrypi/rp2xxx/src/hal/i2c.zig [hunk 2]
```diff
@@ -469,41 +512,49 @@ pub const I2C = enum(u1) {
...
+    ///       in a separate memory.
+    ///
+    pub fn writev_then_readv_blocking(i2c: I2C, addr: Address, write_chunks: []const []const u8, read_chunks: []const []u8, timeout: ?time.Duration) TransactionError!void {
+        if (addr.is_reserved())
+            return TransactionError.TargetAddressReserved;
+
+        const write_vec = microzig.utilities.Slice_Vector([]const u8).init(write_chunks);
+        const read_vec = microzig.utilities.Slice_Vector([]u8).init(read_chunks);
+
+        if (write_vec.len() == 0)
```
Comment (haydenridd) [inline port/raspberrypi/rp2xxx/src/hal/i2c.zig:538]: Also needs to be `size()`!

**Review by haydenridd — CHANGES_REQUESTED** [review]
Just ran it on my hardware needed to change those two things to get it to compile. Runs fine though with my existing "self test" routines though so good to go after that!

**Review by haydenridd — APPROVED** [review]
Sweet, worked like a charm!

**Review by mattnite — APPROVED** [review]

**Comment by mattnite** [comment]
Looks great @ikskuh ! love the deadlines

**Pull request merged**